### PR TITLE
moved slice by class id before shape matching

### PR DESCRIPTION
**Pull request by simian-typist**
Addressing issue https://github.com/tensorflow/tensorflow/issues/30983

If you passed sample weights and a class id to a keras metric it would
enforce that the predictions, labels and weights had the same shape
before slicing the labels and predictions, resulting in a shape mismatch
when trying to apply the weights later.

### CODE REVIEW

**Comment by simian-typist** [comment]
I signed it!

**Comment by rthadur** [comment]
Closing this PR as this not against `master`, please open a new PR against `master` 
CC @mihaimaruseac
